chalicelib/checks/audit_checks.py

```diff
@@ -190,6 +191,66 @@ def expset_opf_unique_files_in_experiments(connection, **kwargs):
     return check
 
 
+@check_function(days_back=14)
+def expset_opf_unique_files(connection, **kwargs):
+    '''
+    Checks Experiments and Experiment Sets with other_processed_files and looks
+    if any opf is also present within the raw, processed or reference files.
+    '''
+    check = CheckResult(connection, 'expset_opf_unique_files')
+    days_back = kwargs.get('days_back')
+    from_date_query, from_text = wrangler_utils.last_modified_from(days_back)
+    # opfs can be on Exps or ExpSets: search ExpSets for each case and merge results
+    opf_query = ('other_processed_files.files.uuid%21=No+value' +
+                 from_date_query +
+                 '&field=experiments_in_set&field=other_processed_files')
+    opf_sets = ff_utils.search_metadata(
+        'search/?type=ExperimentSet&' + opf_query, key=connection.ff_keys)
+    opf_exps = ff_utils.search_metadata(
+        'search/?type=ExperimentSet&experiments_in_set.' + opf_query, key=connection.ff_keys)
+    # merge
+    es_list = [expset['@id'] for expset in opf_sets]
+    for expset in opf_exps:
+        if expset['@id'] not in es_list:
+            opf_sets.append(expset)
+
+    errors = {}
+    for expset in opf_exps:
+        # skip sets without experiments
+        if not expset.get('experiments_in_set'):
+            continue
+
+        opfs_id = []  # list opfs @id in the ExpSet and Exps
+        all_files = []  # list all raw, processed and reference files in the ExpSet
+        if expset.get('other_processed_files'):
+            opfs_id.extend([f['@id'] for grp in expset['other_processed_files'] for f in grp.get('files', [])])
+        for exp in expset['experiments_in_set']:
+            if exp.get('other_processed_files'):
+                opfs_id.extend([f['@id'] for grp in exp['other_processed_files'] for f in grp.get('files', [])])
+            all_files.extend([f['@id'] for f in exp.get('files', [])])
+            all_files.extend([f['@id'] for f in exp.get('processed_files', [])])
+            all_files.extend([f['@id'] for f in exp.get('reference_files', [])])
+        all_files.extend([f['@id'] for f in expset.get('processed_files', [])])
```

Comment:
if you're using `field=experiments_in_set&field=other_processed_files` in the search query, then it seems that processed_files won't be in the search result, no?

Reply:
Good catch! Fixed now